### PR TITLE
Remove the EntityFrameworkCore submodule

### PR DESCRIPTION
**Pull request by natemcmaster**
Resolves https://github.com/aspnet/AspNetCore/issues/3869

This removes EntityFrameworkCore as a git submodule. This repo will build on its own, and this repo will consume EFCore via PackageReference.

FYI @Eilon

### CODE REVIEW

File: build/CodeSign.props [hunk 1]
```diff
@@ -43,6 +43,13 @@
       <FilesToSign Include="Microsoft.Extensions.Options.DataAnnotations.dll"               Certificate="$(AssemblySigningCertName)" Container="Microsoft.AspNetCore.App" />
       <FilesToSign Include="Microsoft.Extensions.Options.dll"                               Certificate="$(AssemblySigningCertName)" Container="Microsoft.AspNetCore.App" />
       <FilesToSign Include="Microsoft.Extensions.Primitives.dll"                            Certificate="$(AssemblySigningCertName)" Container="Microsoft.AspNetCore.App" />
+    <!-- These files came from the aspnet/EntityFrameworkCore build, but have to be re-signed because we crossgen them. -->
+      <FilesToSign Include="Microsoft.EntityFrameworkCore.Abstractions.dll"                 Certificate="$(AssemblySigningCertName)" Container="Microsoft.AspNetCore.App" />
```
Comment (smitpatel) [inline build/CodeSign.props]: Would analyzers be in this list?

File: build/dependencies.props [hunk 2]
```diff
@@ -139,7 +148,6 @@
     <MicrosoftApplicationInsightsAspNetCorePackageVersion>2.1.1</MicrosoftApplicationInsightsAspNetCorePackageVersion>
     <MicrosoftAspNetIdentityEntityFrameworkPackageVersion>2.2.1</MicrosoftAspNetIdentityEntityFrameworkPackageVersion>
     <MicrosoftAspNetWebApiClientPackageVersion>5.2.6</MicrosoftAspNetWebApiClientPackageVersion>
-    <MicrosoftAzureDocumentDBCorePackageVersion>1.7.1</MicrosoftAzureDocumentDBCorePackageVersion>
```
Comment (smitpatel) [inline build/dependencies.props]: While this package versions are going away from here, how would EF Core control package versions in future?

**Review by smitpatel — APPROVED** [review]
LGTM.

Commented a few questions in order to understand if there would be any changes on EF Core currently builds.

File: build/CodeSign.props (hunk 1 above)
Comment (natemcmaster) [reply build/CodeSign.props]: I don't think so. Analyzers shouldn't be in the shared framework.

File: build/dependencies.props (hunk 2 above)
Comment (natemcmaster) [reply build/dependencies.props]: However you want :) Think of EFCore as a sovereign repo now. You are welcome to keep using the KoreBuild conventions like build/dependencies.props, or you can do something else.

This also means aspnetci won't push changes to the master branch with dependency updates. The recommended replacement for this is the .NET Core 3.0 dependency flow tools. See https://github.com/dotnet/arcade/blob/master/Documentation/DependenciesFlowPlan.md

Comment (Eilon) [reply build/dependencies.props]: I highly urge most repos to stick to the common patterns, though.

**Pull request merged**